src/coreclr/jit/simd.cpp

```diff
@@ -1212,7 +1211,8 @@ GenTree* Compiler::impSIMDPopStack(var_types type, bool expectAddr, CORINFO_CLAS
     // SIMD type struct that it points to.
     if (expectAddr)
     {
-        assert(tree->TypeGet() == TYP_BYREF);
+        assert(tree->TypeIs(TYP_BYREF, TYP_I_IMPL));
```

Comment:
Yes. If you have say a `Vector4*` rather than a `ref Vector4`, then this assert could fire.
This wasn't firing for the older SIMD logic because there was explicit validation that the first parameter was `TYP_BREF` and it would say "unsupported" otherwise.

However, there is no reason to block `Vector4*` and we can simply support it instead.